src/ssh/test-sshbridge.c

```diff
@@ -253,6 +258,30 @@ setup (TestCase *tc,
     host = g_strdup ("127.0.0.1");
   argv[1] = host;
 
+  /* run our tests with temp home dir, to avoid influence from the real ~/.ssh */
+  tc->home_dir = g_dir_make_tmp ("home.XXXXXX", NULL);
+  g_assert (tc->home_dir != NULL);
+  env = g_environ_setenv (env, "HOME", tc->home_dir, TRUE);
+
+  if (fixture && fixture->knownhosts_home)
+    {
+      gchar *content;
+
+      tc->home_ssh_dir = g_build_filename (tc->home_dir, ".ssh", NULL);
+      g_assert (tc->home_ssh_dir != NULL);
+      tc->home_knownhosts_file = g_build_filename (tc->home_ssh_dir, "known_hosts", NULL);
+      g_assert (tc->home_knownhosts_file != NULL);
+      g_assert_cmpint (mkdir (tc->home_ssh_dir, 0700), ==, 0);
```

Comment:
Any reason not to use `g_mkdir` ? Not that we're running these on windows any time soon, more a general question.

Reply:
Not a strong one, really. I think it would just need a different #include file.